### ISSUE TITLE
Stratum Authentication Failed

### ISSUE THREAD

**Issue by stestaz** [zone117x/node-cryptonote-pool#107]
Hi, I've correctly started my pool but I cannot connect from client, I receive a Stratum Authentication failed error on it an nothing on the server.

**Comment by gaffelosked** [comment]
That's because you're using a invalid wallet address, I'm 100% certain.

**Comment by zanteweb** [comment]
Hi, I have the same problem. wallet address is 100% valid!!

**Comment by software4u2016** [comment]
Any news??

**Comment by cibermax** [comment]
Any news??

**Comment by CanadianRepublican** [comment]
i have this problem

**Comment by zone117x** [comment]
Post more information about the problem such as: the wallet address, is monero or an alt being used, what clients are you using, etc.

Come on guys...

**Comment by CanadianRepublican** [comment]
url to problem with images and description https://github.com/fancoder/cryptonote-universal-pool/issues/35

**Comment by cibermax** [comment]
Hi, Sorry =)
1. Program (Linux Claymore xmr miner ) https://drive.google.com/open?id=0B69wv2iqszefZzNWX05Ub0IwRVk

2. Wallets all (node address, poloniex.com wallet addres and paymant id)
![image](https://cloud.githubusercontent.com/assets/7147503/22000216/fe4a383e-dc4e-11e6-862e-47e92eb1e6d2.png)
![image](https://cloud.githubusercontent.com/assets/7147503/22000229/1a417de0-dc4f-11e6-975d-294470fff4eb.png)

3. node-cryptonote-pool console no errors no miners
![image](https://cloud.githubusercontent.com/assets/7147503/22000119/8a89eb4c-dc4e-11e6-8162-af6c16be0950.png)

**Comment by zone117x** [comment]
It looks like the pool does not support validating address formats with the payment ID integrated. 
Pull requests to support them are welcome.

**Comment by zone117x** [comment]
@snipa22 may be able to implement a fix

**Comment by Snipa22** [comment]
Integrated address information I'll need later Yes, src/cryptonote_core/cryptonote_basic_impl.cpp, get_account_integrated_address_from_str

**Comment by trampk** [comment]
how much will it cost  the payment ID integrated in your pool? or advise anyone can do it...

**Comment by Snipa22** [comment]
There's no expected price at this time, I've got a working copy of a pool w/ payment ID integration done, but it's not OSS.

**Comment by Snipa22** [comment]
This has been reviewed, part of the code is complete for this, but there's a fair bit of hacking that I need to do at the payment code to get this working offhand.  The current CN pool is not particularly designed for this.

**Comment by zone117x** [comment]
Interested parties should set up a fund for @Snipa22 to get it working in this pool and merge it.

**Comment by Snipa22** [comment]
As a quick touch on, going down this path will also unlock the availability for exchange payouts, both of these systems are fairly identical at this point, and will fix a couple of the other login-related bugs that are out there.

**Comment by gabrieldodan** [comment]
I have similar issue. Is there any work around? Thanks.

**Comment by CanadianRepublican** [comment]
I managed to take a different pool version and get it working ..

I will make a repository later today 

http://monero.hiive.biz

**Comment by Snipa22** [comment]
What version of NodeJS are you guys using for this?  It occurred to me that the check in cryptonote-utils itself may not be working, and I might need to go patch that.

**Comment by gabrieldodan** [comment]
I use v0.10.48 

I have tried to use last stable version of node but npm update fails. 

@CanadianRepublican let me know please when the repository is done

Thanks

**Comment by Snipa22** [comment]
Thanks!  I'll patch a local copy and see.  I think I know what might be happening.

**Comment by Snipa22** [comment]
Son of a gun.  Okay, I have the bug now.  I'll have a pull request into the other lib shortly.  Sorry guys, I assumed that node-cryptonote-util was working properly.

**Comment by CanadianRepublican** [comment]
https://github.com/CanadianRepublican/monero-universal-pool

give me an hour to confirm this works. ( it should all be there now )

as for the npm update

look at the package.json 

Change line  with  "cryptonote-util" to

"cryptonote-util": "git://github.com/moneromooo/node-cryptonote-util.git#rct"

Or you can just copy the package.json from my repository.

**Comment by Snipa22** [comment]
You'll probally want an even more updated version, gimme a few and I'll have a patch into the upstream to fix this.  I think I know where the bug is, I recently fixed it on mine.

**Comment by Snipa22** [comment]
Moneromoo's doesn't fix it, I'll have a merge request in, in 5 that patches to my version.

**Comment by CanadianRepublican** [comment]
my pool is working with that change..

I will install another VM and rerun the install.

In case it helps .. I used node-v0.10.48-linux-x64 on Ubuntu

**Comment by Snipa22** [comment]
His code is returning the same buffers that the current codebase is.

Pull request: https://github.com/zone117x/node-cryptonote-pool/pull/142 should fix this.

**Comment by Snipa22** [comment]
Thanks for the merge!

```
snipa@iof601:~/test_node$ node base/init.js ; node moo/init.js; node snipa/init.js
false
<SlowBuffer 00 00 00 00 00 00 00 12 f1 df 97 8e 58 64 23 08 f4 9d c0 6c 64 58 7e b0 f9 c7 c3 77 1e 94 28 ae fa c5 b8 c3 f2 ed 50 4a a6 94 d5 09 3b 5c f9 a2 cb 22 b6 ...>
<SlowBuffer 00 00 00 00 00 00 00 12 9d aa 30 e5 40 76 f5 82 53 87 c1 8d bd 88 09 f7 c9 ad b3 0e d2 81 07 5e 3b 3b 33 e9 49 f7 9e 64 69 38 c7 aa 30 45 80 56 be 73 68 ...>
false
<SlowBuffer 00 00 00 00 00 00 00 12 f1 df 97 8e 58 64 23 08 f4 9d c0 6c 64 58 7e b0 f9 c7 c3 77 1e 94 28 ae fa c5 b8 c3 f2 ed 50 4a a6 94 d5 09 3b 5c f9 a2 cb 22 b6 ...>
<SlowBuffer 00 00 00 00 00 00 00 12 9d aa 30 e5 40 76 f5 82 53 87 c1 8d bd 88 09 f7 c9 ad b3 0e d2 81 07 5e 3b 3b 33 e9 49 f7 9e 64 69 38 c7 aa 30 45 80 56 be 73 68 ...>
true
18
18
snipa@iof601:~/test_node$ node -v
v0.10.32
```

**Comment by CanadianRepublican** [comment]
Does this merge also fix the Stratum Authentication failed error.

The previously mentioned fixes didn't work for me (not your ones from today). That's actually why I hacked together another pool from this one and cryptonote-universal-pool.

https://github.com/CanadianRepublican/monero-universal-pool

**Comment by CanadianRepublican** [comment]
oh and feel free to add the systemd files from monero daemon and wallet to your repository if you want ..

**Comment by Snipa22** [comment]
Stratum Authentication Failed is a generic error from your miner.  This should fix it IF the bug is in relation to the address verification portions.  This WILL not fix it if you are attempting to use integrated addresses.  My new pool source is aimed at fixing the latter of the two of these, once the FFS is funded, we should be releasing.

**Comment by CanadianRepublican** [comment]
+Snipa22 

I just did the install from start to finish and I can confirm its working with 
Ubuntu 16.04 and Node 0.10.48. 

I have attached a text file with the command I used to install . Be warned it is not a bash script, but it does include instructions for the install of Redis and Node. It does not include the Apache/Nginx portions

[zone117x_install.txt](https://github.com/zone117x/node-cryptonote-pool/files/752667/zone117x_install.txt)

**Comment by Snipa22** [comment]
Glad to hear it!  Sorry about that bug.  I've been working on a huge chunk of new source, and when I went back, it was loading my fixed copy of the cn-utils, rather than the old ones.

Not a bad looking setup script, I'm working on a massive deploy script for my new pool, it's not super-complex, but it has alot of moving parts. :)

**Comment by gabrieldodan** [comment]
Works now. Thanks guys!!!

Any possibility to run the pool using the last stable version of node? I have tried "cryptonote-util": "git://github.com/moneromooo/node-cryptonote-util.git#rct" but doesn;t works

Regards

**Comment by CanadianRepublican** [comment]
its true .. moneromooo only seems to work on my pool.

**Comment by Snipa22** [comment]
No, this code base does not work w/ the latest stable version of node.  It needs a fairly heavy overall to pull that off.  The cnUtils from my branches do work w/ latest version (6.9.2), as thats what I targeted for my new pool code base.

**Comment by Ping-Hosting** [comment]
Hi, 

I have the same problem, impossible to use my Wallet Address of BCN to login;

{"id":1,"jsonrpc":"2.0","error":{"code":-1,"message":"invalid address used for login"}}

Tested with CPUMiner , how to fix this problem?

**Comment by gjnardoni** [comment]
Me too with BCN and BitCoal  (also Quazarcoin using forknote-pool)
